src/components/TableComponent/index.jsx

```diff
@@ -0,0 +1,182 @@
+      if (ordering !== prevOrdering) {
+        this.props.sortTable(ordering);
+      } else if (page !== prevPage) {
+        this.props.paginateTable(parseInt(page, 10));
+      }
+    }
+  }
+
+  componentWillUnmount() {
+    this.props.clearTable();
+  }
+
+  renderTableContent() {
+    const {
+      className,
+      currentPage,
+      pageCount,
+      tableSortable,
+      data,
+      ordering,
+      formatData,
+      id,
+      loading,
+    } = this.props;
+
+    const columnConfig = this.props.columns.map(column => ({
+      ...column,
+      onSort: column.columnSortable ? direction => updateUrl({
+        page: 1,
+        ordering: direction === 'desc' ? `-${column.key}` : column.key,
+      }) : null,
+    }));
+
+    let sortDirection;
+    let sortColumn;
+
+    if (tableSortable) {
+      sortDirection = ordering && ordering.indexOf('-') !== -1 ? 'desc' : 'asc';
+      sortColumn = (ordering && ordering.replace('-', '')) || columnConfig[0].key;
+    }
+
+    return (
+      <div className={className}>
+        <div className="row">
+          <div className="col">
+            {loading && <TableLoadingOverlay />}
+            <div className="table-responsive">
+              <Table
+                id={id}
+                className={['table-sm', 'table-striped']}
+                columns={columnConfig}
+                data={formatData(data)}
+                tableSortable={tableSortable}
+                defaultSortedColumn={sortColumn}
+                defaultSortDirection={sortDirection}
+              />
+            </div>
+          </div>
+        </div>
+        <div className="row mt-2">
+          <div className="col d-flex justify-content-center">
+            <Pagination
+              paginationLabel={`${id}-pagination`}
+              pageCount={pageCount}
+              currentPage={currentPage}
+              onPageSelect={page => updateUrl({ page })}
+            />
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  renderLoadingMessage() {
+    return <LoadingMessage className="table-loading" />;
+  }
+
+  renderErrorMessage() {
+    return (
+      <StatusAlert
+        alertType="danger"
+        iconClassNames={['fa', 'fa-times-circle']}
+        title="Unable to load data"
+        message={`Try refreshing your screen (${this.props.error.message})`}
+      />
+    );
+  }
+
+  renderEmptyDataMessage() {
+    return (
+      <StatusAlert
+        alertType="warning"
+        iconClassNames={['fa', 'fa-exclamation-circle']}
+        message="There are no results."
+      />
+    );
+  }
+
+  render() {
+    const {
+      data,
+      loading,
+      error,
+    } = this.props;
+
+    return (
+      <React.Fragment>
+        {error && this.renderErrorMessage()}
+        {loading && !data && this.renderLoadingMessage()}
+        {!loading && !error && data && data.length === 0 &&
+          this.renderEmptyDataMessage()
+        }
```

Comment:
I guess this comes down to preference but the first 3 render functions return very small jsx elements so I would advocate for just moving them directly into the render function.